src/current/v24.2/cost-based-optimizer.md

```diff
@@ -277,27 +277,74 @@ Only tables with `ZONE` [survivability]({% link {{ page.version.version }}/multi
 
 ## Query plan cache
 
-CockroachDB uses a cache for the query plans generated by the optimizer. This can lead to faster query execution since the database can reuse a query plan that was previously calculated, rather than computing a new plan each time a query is executed.
+CockroachDB caches some of the query plans generated by the optimizer. The query plan cache is used for the following types of statements:
+
+- Prepared statements.
+- Non-prepared statements using identical constant values.
+
+Caching query plans leads to faster query execution: rather than generating a new plan each time a query is executed, CockroachDB reuses a query plan that was previously generated.
 
 The query plan cache is enabled by default. To disable it, execute the following statement:
 
 {% include_cached copy-clipboard.html %}
 ~~~ sql
-> SET CLUSTER SETTING sql.query_cache.enabled = false;
+SET CLUSTER SETTING sql.query_cache.enabled = false;
 ~~~
 
-Only the following statements use the plan cache:
+The following statements can use the plan cache: [`SELECT`]({% link {{ page.version.version }}/select-clause.md %}), [`INSERT`]({% link {{ page.version.version }}/insert.md %}), [`UPDATE`]({% link {{ page.version.version }}/update.md %}), [`UPSERT`]({% link {{ page.version.version }}/upsert.md %}), and [`DELETE`]({% link {{ page.version.version }}/delete.md %}).
 
-- [`SELECT`]({% link {{ page.version.version }}/select-clause.md %})
-- [`INSERT`]({% link {{ page.version.version }}/insert.md %})
-- [`UPDATE`]({% link {{ page.version.version }}/update.md %})
-- [`UPSERT`]({% link {{ page.version.version }}/upsert.md %})
-- [`DELETE`]({% link {{ page.version.version }}/delete.md %})
+Two types of plans can be cached: custom and generic. Refer to [Query plan type](#query-plan-type).
 
-The optimizer can use cached plans if they are:
+### Query plan type
 
-- Prepared statements.
-- Non-prepared statements using identical constant values.
+The following types of plans can be cached:
+
+- *Custom* query plans are generated for a given query structure and optimized for specific placeholder values, and are re-optimized on subsequent executions. By default, the optimizer uses custom plans.
+- {% include_cached new-in.html version="v24.2" %} *Generic* query plans are generated and optimized once without considering specific placeholder values, and are **not** regenerated on subsequent executions, unless the plan becomes stale due to [schema changes]({% link {{ page.version.version }}/online-schema-changes.md %}) or new [table statistics](#table-statistics) and must be re-optimized. Once a generic plan is optimized for a given statement, the CPU utilization associated with query planning for all subsequent executions of that statement should be 0%.
```

Comment:
test

Reply:
Ugh sorry. Github has lost my comment on this line twice now, so I've had to retype it:

> Once a generic plan is optimized for a given statement, the CPU utilization associated with query planning for all subsequent executions of that statement should be 0%.

I'd prefer to say something like: "Once a generic plan is optimized for a given statement, it will not be reoptimized during subsequent executions, eliminating most of the query latency attributed to planning."

Reply:
I simplified to `This eliminates most of the query latency attributed to planning.` since I think most of that line is redundant with the text that precedes it. Let me know if I got this wrong.